src/encoder/ffmpeg/avcontext.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "disable_warnings_push.hpp"
+extern "C" {
+#include <libavcodec/avcodec.h>
```

Comment:
Either remove this include and forward declare `AVCodec` or remove forward declaration of `AVCodecContext`